File: python/7 kyu/421.py
# https://www.codewars.com/kata/5a1e6323ffe75f71ae000026/train/python

# Passed

def is_turing_equation(s):
    get_proper_num = lambda num: int(num[::-1])
    first_num, second_num, total_sum = list(map(get_proper_num, s.replace("=", "+").split("+")))
    result = first_num + second_num == total_sum
    return result

output = is_turing_equation('73+42=16')
print(output)

File: nodejs/7 kyu/421.js
// https://www.codewars.com/kata/59377c53e66267c8f6000027/train/javascript

// Passed

function alphabetWar(fight) {
  let leftSide = {
    w: 4,
    p: 3,
    b: 2,
    s: 1,
  };

  let rightSide = {
    m: 4,
    q: 3,
    d: 2,
    z: 1,
  };

  let [leftSideSum, rightSideSum] = [0, 0];
  for (const letter of fight) {
    if (letter in leftSide) {
      leftSideSum += leftSide[letter];
    } else if (letter in rightSide) {
      rightSideSum += rightSide[letter];
    }
  }

  let result;
  if (leftSideSum === rightSideSum) {
    result = "Let's fight again!";
  } else if (leftSideSum > rightSideSum) {
    result = "Left side wins!";
  } else {
    result = "Right side wins!";
  }

  return result;
}

const output = alphabetWar("zdqmwpbs");
console.log(output);